### PR TITLE
Feature: Adding Angular Velocity to Forward Kinematics

### PR DESCRIPTION
**Pull request by mattking-smith**
# Feature Purpose
This adds in the angular velocity into the forward kinematics computation.

## Issues Resolved
Merging this branch will resolve issue #20.

### Features Added
Verified C++ unit tests

### CODE REVIEW

File: gpmp2/kinematics/tests/testArm.cpp [hunk 1]
```diff
@@ -296,13 +304,13 @@ TEST(Arm, WAMexample) {
...
   pvec_exp.push_back(Point3(-0.0066, 0.7134, -0.0576));
 
-  vvec_exp.push_back(Vector3(0, 0, 0));
-  vvec_exp.push_back(Vector3(0, 0, 0));
-  vvec_exp.push_back(Vector3(-0.0557, 0.0204, -0.0511));
-  vvec_exp.push_back(Vector3(-0.0606, 0.0234, -0.0595));
-  vvec_exp.push_back(Vector3(-0.0999, -0.0335, -0.0796));
-  vvec_exp.push_back(Vector3(-0.0999, -0.0335, -0.0796));
-  vvec_exp.push_back(Vector3(-0.1029, -0.0333, -0.0976));
+  vvec_exp.push_back(Vector6(0, 0, 0, 0, 0, 0.1));
```
Comment (varunagrawal) [inline gpmp2/kinematics/tests/testArm.cpp:307]: Please add a high level comment where you got these numbers from and how to reproduce them.

**Review by varunagrawal — CHANGES_REQUESTED** [review]

File: gpmp2/kinematics/tests/testArm.cpp [hunk 2]
```diff
@@ -213,7 +218,8 @@ TEST(Arm, 2linkPlanarExamples) {
   }
 }
 
-/* ************************************************************************** */
+/* **************************************************************************
+ */
```
[the remark this answers is not in the document]
Comment (mattking-smith) [reply gpmp2/kinematics/tests/testArm.cpp:222]: Yeah it looks like autoformatting is killing me here for some reason.

[the remark this answers is not in the document]
Comment (mattking-smith) [reply gpmp2/kinematics/tests/testArm.cpp:222]: Not auto formatting getting in the way. I think this was me not noticing from commenting out and then recommenting in the tests.

**Review by varunagrawal — APPROVED** [review]
LGTM

**Pull request merged**